module/command_tree_generator.py

```diff
@@ -29,25 +29,28 @@ def add_frame_commands(root_node, mod=common.frames) -> None:
 
     # add all commands to root node
     for command in commands:
+        # indexes everything of the frame name after 'Frame'
         name = command[0][5:]
+
+        # get values from the frame class
         parameters = inspect.getfullargspec(command[1].set_data).annotations
         description = command[1].DESCRIPTION
+
         add_command(current_node, name, parameters, description)
 
 
 def add_command_from_json(json_command, root_node, prohibited_words) -> None:
     """
     add one json command to root node
     """
-    if prohibited_words:
-        prohibited_keywords = set().union(prohibited_words)
+
     json_command["target"] = json_command["target"].upper()
     current_node = root_node
 
     # Per path checking if it has a child
     json_command["category"] = json_command["category"].upper()
     for path_piece in json_command["category"].split(" "):
-        if prohibited_words and path_piece in prohibited_keywords:
+        if prohibited_words and path_piece in set(prohibited_words):
```

Comment:
This is less efficient as you will generate a new set every time you go through this loop